src/xkbcomp/compat.c

```diff
@@ -668,18 +668,18 @@ HandleInterpBody(CompatInfo *info, VarDef *def, SymInterpInfo *si)
     ExprDef *arrayNdx;
 
     for (; def; def = (VarDef *) def->common.next) {
-        if (def->name && def->name->expr.op == EXPR_FIELD_REF) {
+        ok = ExprResolveLhs(info->ctx, def->name, &elem, &field, &arrayNdx);
+        if (!ok)
+            continue;
+        if (elem) {
```

Comment:
This also puts out the error on array refs with element (`x.y[z] = foo`), but that's a good thing, seems like the `x` was was just completely ignored in this case (though would still fail in `SetInterpField` due to the array index).

Reply:
Well, I have to confess I missed this! 😳 Thank you for pointing it. I checked again that this does not further have side effects. I think we are good, but I will wait a few days before merging.